app/services/ProductDAO.scala

```diff
@@ -29,24 +31,30 @@ trait ProductDAO {
       .unsafeRunSync()
   }
 
-  /** Get a specific product from the inventory
+  /** Get a specific product from the inventory, if an error is caught (meaning the uuid string is incorrect)
+   * then return null
    *
-   * @param id
-   * @tparam T
+   * @param id: String of the UUID given by the user
+   * @tparam T: What the query will put its data in the format of.
    * @return
    */
   def getResultById[T: Read](id: String) : List[T] = {
-
-    sql"select * from Product where id=${id}::uuid "
-      .query[T]
-      .to[List]
-      .transact(xa)
-      .unsafeRunSync()
+    val status = Try(
+      sql"select * from Product where id=$id::uuid "
+        .query[T]
+        .to[List]
+        .transact(xa)
+        .unsafeRunSync()
+    )
+    status match{
+      case Success(list) => list
+      case Failure(_) => null
```

Comment:
do not use nulls in Scala (or java, or really anywhere). If there is a failure, based on your return value you should return an empty list and log the error

Reply:
Good to know, will do!